extension/healthcheckextension/healthcheckextension_test.go

```diff
@@ -71,6 +72,45 @@ func TestHealthCheckExtensionUsageWithoutCheckCollectorPipeline(t *testing.T) {
 	require.Equal(t, http.StatusServiceUnavailable, resp2.StatusCode)
 }
 
+func TestHealthCheckExtensionUsageWithCustomizedPathWithoutCheckCollectorPipeline(t *testing.T) {
+	config := Config{
+		TCPAddr: confignet.TCPAddr{
+			Endpoint: testutil.GetAvailableLocalAddress(t),
+		},
+		CheckCollectorPipeline: defaultCheckCollectorPipelineSettings(),
+		Path:                   "/health",
+	}
+
+	hcExt := newServer(config, zap.NewNop())
+	require.NotNil(t, hcExt)
+
+	require.NoError(t, hcExt.Start(context.Background(), componenttest.NewNopHost()))
```

Comment:
You should probably override the hcExt.server with a `httptest.Server`, which should probably be better for this case.

Reply:
Sorry I afraid I do not fully understand what do you mean here, so I will post my understand so you can clarify. 

Do you mean I should create one `httptest.Server` based on the `hcExt.Handler` by doing something like the following
```go
	hcExt := newServer(config, zap.NewNop())
	server := httptest.NewServer(hcExt.server.Handler)
```
then in the tests, send request to `server.URL` instead. 

Or do you mean I should repalce `hcExt`with one `httptest.Server` entirely and send request to it? If that is the case, I afraid that is not feasible. As that is the health check extension server and we should test against it. 

Please correct me if I got something wrong. Thanks

Reply:
The first one, but it would be to better control the lifecycle of the server, to avoid flaky tests. If you are using "assert.Eventually", you might not need to use the httptest here.